src/qt/intro.cpp

```diff
@@ -128,14 +126,22 @@ Intro::Intro(QWidget *parent, uint64_t blockchain_size, uint64_t chain_state_siz
     );
     ui->lblExplanation2->setText(ui->lblExplanation2->text().arg(PACKAGE_NAME));
 
-    int64_t prune_target_mib = std::max<int64_t>(0, gArgs.GetArg("-prune", 0));
-    if (prune_target_mib > 1) { // -prune=1 means enabled, above that it's a size in MiB
+    connect(ui->prune, &QCheckBox::toggled, [this](bool prune_checked) {
+        UpdatePruneLabels(prune_checked);
+        UpdateFreeSpaceLabel();
+    });
+
+    int64_t prune_target_mib = gArgs.GetArg("-prune", 0);
+    if (prune_target_mib * 1024 * 1024 >= MIN_DISK_SPACE_FOR_BLOCK_FILES) {
```

Comment:
> The two versions are equivalent because of [other code forbidding values from 2 to 549](https://github.com/bitcoin/bitcoin/blob/6196e930018181301b5972842ae384ea4288ff34/src/init.cpp#L1061-L1063)).

That is not the case.
The code you mentioned is called much later:
https://github.com/bitcoin/bitcoin/blob/e7f84503571c171a7e6728cd2d77dd4103bd7a6f/src/qt/bitcoin.cpp#L581
after
https://github.com/bitcoin/bitcoin/blob/e7f84503571c171a7e6728cd2d77dd4103bd7a6f/src/qt/bitcoin.cpp#L491

It implies that forbidding values from 2 to 549 are accepted and
```
$ bitcoin-qt -choosedatadir -prune=42
```
works as `prune=1`.

This PR ignores prune forbidding values from 2 to 549.

Move it to a separate _non_-refactoring commit?

Reply:
> Note: I found the change on this line confusing, and might suggest reverting...

Reverted.